### ISSUE TITLE
ReferenceError unless variable is passed to console.log()

### ISSUE THREAD

**Issue by AnyhowStep** [TypeStrong/ts-node#591]
I am having this weird problem where I have, roughly, the following code snippet,

```
import * as s from "../../../some-lib";

export class SomeClass extends OtherClass {
    public constructor () {
        super(s.someVar);
    }
}
```

`ts-node` gives me: `ReferenceError: s is not defined`

```
import * as s from "../../../some-lib";
console.log(s);
export class SomeClass extends OtherClass {
    public constructor () {
        super(s.someVar);
    }
}
```

If I just add `console.log(s);`, it prints and runs fine, without `ReferenceError`;

I'm on `ts-node 5.0.1`

It's happening to me for many files, but not all of them.

-----

Edit: I just went through about 20 of my files and added that `console.log()` and `ts-node` doesn't complain anymore. A little concerning I had to do that, though.

------

Edit: I downgraded to TS 2.8.3 and don't seem to be getting that problem anymore. I was on `typescript@2.9.0-dev.20180516`

**Comment by seepel** [comment]
This seems to still be an issue in typescript 2.9.1.

I also have an instance where I only use an import inside the call to super for a constructor and see the same issue.

Placing a break point right after the module is imported I can see in scope an uncompiled(?) module that looks something like this:
```
Module {
  _compile:function (code, fileName) { … }
  children:Array(3) [Module, Module, Module]
  exports:Object {__esModule: true}
  filename:"/path/to/project/some-lib.ts"
  id:"/path/to/project/some-lib.ts"
  loaded:false
  parent:Module {
    id: "/path/to/project/parent.ts", 
    exports: Object, parent: Module, …}
  paths:Array(9) […]
  __proto__:Object {load: , require: , _compile: , …}
```

**Comment by phiresky** [comment]
Same issue here. Possible cause https://github.com/Microsoft/TypeScript/issues/24620

"Fix should be already in typescript@next today, and in typescript@2.9.2 in two weeks."

**Comment by shaunxu** [comment]
I have the exactly same issue when using `ts-node` with `mocha`. Since from `v6.x` `ts-node` performs type checking by default and in order to bypass some compile errors in my specs, I changed `--require ts-node/register` to `--require ts-node/register/transpile-only`. But when I upgrade to TypeScript 2.9.1 I got this error.

**Comment by blakeembrey** [comment]
Can someone let me know if this was resolved with `2.9.2`?

**Comment by seepel** [comment]
Sorry for the late reply, but yes everything seems to be working correctly with `2.9.2` Thank you!
